src/xb-silo-query.c

```diff
@@ -41,11 +45,33 @@ xb_silo_query_node_matches (XbSilo *self,
 	if (section->predicates != NULL) {
 		for (guint i = 0; i < section->predicates->len; i++) {
 			XbStack *opcodes = g_ptr_array_index (section->predicates, i);
-			if (!xb_machine_run (machine, opcodes, result, query_data, error))
+			g_auto(XbValueBindings) predicate_bindings = XB_VALUE_BINDINGS_INIT ();
+			guint predicate_bindings_idx = 0;
+			XbValueBindings *predicate_bindings_ptr = (bindings != NULL) ? &predicate_bindings : NULL;
```

Comment:
this looks a bit magic... maybe just an `if`?

Reply:
As in the following?
```
XbValueBindings *predicate_bindings_ptr = NULL;
if (bindings != NULL)
  predicate_bindings_ptr = &predicate_bindings;
```

Not sure that’s any clearer, but I may be misunderstanding what you mean.

Reply:
ya, i think that's much better. using the `?` always makes my brain hurt and I'm sure it all gets compiled to the same thing anyway.